compiler/rustc_hir/src/def.rs

```diff
@@ -39,6 +39,9 @@ pub enum NonMacroAttrKind {
     Tool,
     /// Single-segment custom attribute registered by a derive macro (`#[serde(default)]`).
     DeriveHelper,
+    /// Single-segment custom attribute registered by a derive macro
+    /// but used before that derive macro was expanded (deprecated).
```

Comment:
This corresponds to the case mentioned in https://github.com/rust-lang/rust/pull/66529, right? E.g.

```rust
#[trait_helper] // The helper attribute is used before it introduced.
                        // Sadly, compiles on stable, supported via hacks.
                        // I plan to make a compatibility warning for this.
#[derive(Trait)]
struct S;
```

Could you open a tracking issue to track progress towards the eventual deprecation of this?

Reply:
The deprecation lint is added in https://github.com/rust-lang/rust/pull/79078, I'll make an issue.

Reply:
Issue - https://github.com/rust-lang/rust/issues/79202.